core/src/main/scala/spinal/core/internals/ComponentEmitterVerilog.scala

```diff
@@ -488,6 +491,22 @@ class ComponentEmitterVerilog(
     }
   }
 
+  def emitEnumParams(): Unit = {
+    for((e,encoding) <- localEnums) {
+      for(element <- e.elements) {
+        localparams ++= s"  localparam ${emitEnumLiteral(element, encoding,"")} = ${idToBits(element, encoding)};\n"
+      }
+    }
+  }
+
+  def idToBits[T <: SpinalEnum](enum: SpinalEnumElement[T], encoding: SpinalEnumEncoding): String = {
+    //      val str    = encoding.getValue(enum).toString(2)
+    val str    = encoding.getValue(enum).toString(10)
```

Comment:
This may overflow the Verilog 32 bits integer for enumeration with onehot encoding and more than ?32? elements i guess ?

Reply:
This function is completely referenced here
https://github.com/SpinalHDL/SpinalHDL/blob/524dbb3d7d3aae8680c579a4592db4c57588aeb8/core/src/main/scala/spinal/core/internals/PhaseVerilog.scala#L170

Is that may overflow too?

Reply:
Hoo my bad, i had a reflex from VHDL where integer are 32 bits only, i guess a verilog literal dxxxxxx is fine. 
I have seen the toString(10) and got triggered XDXD